File: app/src/main/java/com/liusbl/listbinderssample/single/SingleActivity.java
package com.liusbl.listbinderssample.single;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import com.liusbl.listbinderssample.R;

import java.util.Arrays;
import java.util.List;

public class SingleActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_single);
        RecyclerView recyclerView = findViewById(R.id.recyclerView);
        PersonAdapter adapter = new PersonAdapter(R.layout.item_single_person);
        recyclerView.setAdapter(adapter);

        List<Person> list = Arrays.asList(
                new Person("0", "Cartman"),
                new Person("1", "Stan"),
                new Person("2", "Kyle"),
                new Person("3", "Kenny"),
                new Person("4", "Gerald"),
                new Person("5", "Randy"),
                new Person("6", "Butters"),
                new Person("7", "Ike"),
                new Person("8", "Wendy"),
                new Person("9", "Ms. Chokesondick"),
                new Person("10", "Mackey"),
                new Person("11", "PC principal"),
                new Person("12", "Jimmy"),
                new Person("13", "Timmy"),
                new Person("14", "Garrison"),
                new Person("15", "Mr. Slave"),
                new Person("16", "Chef"),
                new Person("17", "Scott Tenorman"),
                new Person("18", "Cartman's mom"),
                new Person("19", "Craig"),
                new Person("20", "Token")
        );
        adapter.setItems(list);
    }
}

File: app/src/main/java/com/liusbl/listbinderssample/single/SingleActivity.kt
package com.liusbl.listbinderssample.single

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.liusbl.listbinders.BinderViewHolder
import com.liusbl.listbinders.SingleViewTypeAdapter
import com.liusbl.listbinderssample.R
import kotlinx.android.synthetic.main.activity_single.*
import kotlinx.android.synthetic.main.item_single_person.view.*

class SingleActivity : AppCompatActivity() {
    private val adapter by lazy { PersonAdapter() }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_single)
        recyclerView.adapter = adapter
        val list = listOf(
            Person("0", "Cartman"),
            Person("1", "Stan"),
            Person("2", "Kyle"),
            Person("3", "Kenny"),
            Person("4", "Gerald"),
            Person("5", "Randy"),
            Person("6", "Butters"),
            Person("7", "Ike"),
            Person("8", "Wendy"),
            Person("9", "Ms. Chokesondick"),
            Person("10", "Mackey"),
            Person("11", "PC principal"),
            Person("12", "Jimmy"),
            Person("13", "Timmy"),
            Person("14", "Garrison"),
            Person("15", "Mr. Slave"),
            Person("16", "Chef"),
            Person("17", "Scott Tenorman"),
            Person("18", "Cartman's mom"),
            Person("19", "Craig"),
            Person("20", "Token")
        )
        adapter.setItems(list)
    }

    private inner class PersonAdapter : SingleViewTypeAdapter<Person>(R.layout.item_single_person) {
        override fun onBind(viewHolder: BinderViewHolder<Person>, item: Person) {
            viewHolder.itemView.nameTextView.alpha = 0f
            viewHolder.itemView.nameTextView.animate()
                .setDuration(500)
                .alpha(1f)
                .start()
            viewHolder.itemView.nameTextView.text = item.name
        }
    }

    companion object {
        fun createIntent(context: Context) = Intent(context, SingleActivity::class.java)
    }
}